### ISSUE TITLE
Question. Why not return a fresh task every run?

### ISSUE THREAD

**Issue by sglyon** [JuliaStats/Lora.jl#60]
I am working on writing a generic Gibbs sampler that shares design principles with the code here. (the WIP is on the [gibbs branch of my fork](https://github.com/spencerlyon2/MCMC.jl/tree/gibbs))

I was wondering why we don't always return a fresh task after running a chain? Specifically, I noticed that a task is stopped before being returned in parallel (`run_serialmc_exit`), but not if you run in serial. Unless I'm mistaken this is just a way to tell Julia that it is ok to serialize the task so it can be returned from another process.

When I wrote a simple random walk metropolis routine I found that it was easier to always spin a new task at the end of running a chain (in serial or in parallel) because that way the task is always ready to resume (again, in serial or in parallel). I found that I could get the task to resume at the same place without losing any samples simply by consuming the task one more time to find a new `model.init` value, spinning a new task with that updated `model`, then making sure that the first `produce` in the sampler was `model.init`.

Is there a reason we don't do that here?
